webapp/components/expression/nodes/literal.js

```diff
@@ -51,35 +51,25 @@ const Literal = (props) => {
       params: { ...literalSearchParams, value: nodeValue },
     }
   )
-  const [items, setItems] = useState([])
 
   const onChangeValue = (val) => {
-    const value = getValue(nodeDefCurrent, val)
+    const value = val && getValue(nodeDefCurrent, val)
     onChange(R.pipe(R.assoc('raw', value), R.assoc('value', value))(node))
   }
 
-  if (literalSearchParams) {
-    useEffect(() => {
-      ;(async () => {
-        if (nodeValue) fetchItem()
-        const itemsUpdate = await loadItems({
-          ...literalSearchParams,
-          value: '',
-        })
-        setItems(itemsUpdate)
-      })()
-    }, [])
-  }
+  // on nodeValue update, if literalSearchParams is passed as prop, fetches the selection item for dropdown
+  useEffect(() => {
+    if (literalSearchParams) fetchItem()
+  }, [nodeValue])
 
   const getRenderer = () => {
     if (literalSearchParams) {
       return (
         <Dropdown
-          items={items}
-          itemsLookupFunction={(value) => loadItems({ ...literalSearchParams, value })}
-          itemKeyProp="key"
-          itemLabelProp="label"
-          onChange={(itm) => itm && onChangeValue(itm.key)}
+          items={(value) => loadItems({ ...literalSearchParams, value })}
```

Comment:
is this a function? for me, the name is kind of weird

Reply:
you can pass either an array or a function